modules/ROOT/pages/policy-ping-federate.adoc

```diff
@@ -8,88 +8,156 @@ endif::[]
-----
-{
-    "uid":"john.doe",
-    "mail":"john.doe@example.com",
-    "scope":["uid","mail","cn","givenName"],
-    "grant_type":"password",
-    "cn":"John Doe",
-    "realm":"/",
-    "token_type":"Bearer",
-    "expires_in":3600,
-    "givenName":"John",
-    "access_token":"fa017a0e-1bd5-214c-b19d-03efe9f9847e"
-}
-----
+---
+X-AGW-uid=john.doe
+X-AGW-mail=john.doe@example.com
+X-AGW-token_type=Bearer
+X-AGW-expires_in=3600
+---
+
+For further processing within the flow, the original unparsed response from the validation endpoint is made available. The information is stored in an authentication object. 
+
+For example, if a `mail` field is returned to the policy, you can access the value of the field by using the following DataWeave 2.0 expression:
+
+`#[authentication.properties.userProperties.mail]`
+
+You can obtain the client ID of the OAuth2 token by using the following DataWeave 2.0 expression:
+
+`#[authentication.principal]`
+
+== Communicating with the Token Validation Endpoint Using a Proxy
+
+You can enable the PingFederate OAuth 2.0 Token Enforcement policy to use the gateway proxy settings by specify the following property when starting the Mule runtime engine: 
+
+`anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>`
+
+When you enable this property, the policy uses the Mule proxy settings, if you have specified the following parameters:
+
+* `anypoint.platform.proxy_host=localhost`
+* `anypoint.platform.proxy_port=8080`
+
+=== Configuring Parameters for Mule Applications
+
+When you apply the policy to your API from the UI, a list of parameters is displayed:
+
+[%header%autowidth.spread,cols="30,40,30"]
+|===
+| Element | Description | Example
+| Scopes | Displays a space-separated list of supported scopes. | `READ`, `WRITE`, `READ and WRITE`
+| Scope Validation Criteria | Determines if the token must contain all defined scopes or just one.
+Values: `Contains all scopes` and `Contains any scope` | 
+If the ‘Contains all scopes’ value is selected, the tokens with the scopes `READ` and `WRITE` are accepted. If the ‘Contains any scopes’ value is selected, the tokens with the scope `READ`, `WRITE`, and `READ AND WRITE` are accepted. 
+| Validate TLS Certificate | Enables third-party authentication server TLS validation. | If selected, the communication with the token validation endpoint uses the TLS protocol.
+| Expose Headers | In a proxy scenario, specifies whether headers must be exposed in the request to the backend. User properties returned by the federation server are prefixed with 'X-AGW-' and sent as headers to the backend. | See the <<leveraging-TVE-information,Leveraging Token Validation Endpoint Information for Mule Applications>> for more information about this element.
+| Skip Client Id Validation | Skips the client application's API contract validation. | If selected, the Step 4 of the diagram in the <<how-this-policy-works,How This Policy Works>> section is skipped.
```

Comment:
```suggestion
| Skip Client Id Validation | Skips the client application's API contract validation. | If selected, Step 4 of the diagram in the <<how-this-policy-works,How This Policy Works>> section is skipped.
```
Not a good idea to refer to the content of one topic in another, because this creates intertopic dependencies.

Reply:
Done! Thanks!